styx-scheduler-service/src/test/java/com/spotify/styx/test/FlyteExecutionTest.java

```diff
@@ -57,4 +59,21 @@ public void testFromUrn() {
     assertThat(flyteExecution.getName(), is("test-from-proto-creation"));
     assertThat(flyteExecution.toUrn(), is("ex:flyte-test:testing:test-from-proto-creation"));
   }
+
+  @Rule
+  public ExpectedException exceptionRule = ExpectedException.none();
+
+  @Test
+  public void testFromUrnMalformedString() {
+    exceptionRule.expect(IllegalArgumentException.class);
+    exceptionRule.expectMessage("Expected 4 parts in URN.");
+    FlyteExecution.fromUrn("ex:flyte-test:testing");
+  }
+
+  @Test
+  public void testFromUrlWrongPrefix() {
+    exceptionRule.expect(IllegalArgumentException.class);
+    exceptionRule.expectMessage("Expected URN to start with ex.");
+    FlyteExecution.fromUrn("lp:flyte-test:testing:test-from-url-wrong-prefix");
+  }
```

Comment:
In recent versions of junit `ExpectedException.none()` have been deprecated in favor of using `Assert.assertThrows()`
```suggestion

  @Test
  public void testFromUrnMalformedString() {
    var exception = Assert.assertThrows(
        IllegalArgumentException.class,
        () -> FlyteExecution.fromUrn("ex:flyte-test:testing")
    );

    assertThat(exception.getMessage(), is("Expected 4 parts in URN."));
  }

  @Test
  public void testFromUrlWrongPrefix() {
    var exception = Assert.assertThrows(
        IllegalArgumentException.class,
        () -> FlyteExecution.fromUrn("lp:flyte-test:testing:test-from-url-wrong-prefix")
    );

    assertThat(exception.getMessage(), is("Expected URN to start with ex."));
  }
```

Reply:
assertThrows is introduced in junit5. 
The project is still o junit4 https://github.com/spotify/styx/blob/master/pom.xml#L474
I can look into it another time